x-pack/plugins/security_solution/server/lib/entity_analytics/risk_score/calculate_risk_scores.ts

```diff
@@ -299,22 +342,44 @@ export const calculateRiskScores = async ({
     }
     const identifierTypes: IdentifierType[] = identifierType ? [identifierType] : ['host', 'user'];
 
+    const query = isAlertSamplingEnabled
+      ? {
+          function_score: {
+            query: {
+              bool: {
+                filter,
+                should: [
+                  {
+                    match_all: {}, // This forces ES to calculate score
+                  },
+                ],
+              },
+            },
+            field_value_factor: {
+              field: 'kibana.alert.risk_score', // sort fields by risk score
```

Comment:
Can this be modified to also account for the logic here? https://github.com/elastic/kibana/blob/2c7772b7a3361c2b0262575d57394f9005c5719a/x-pack/plugins/security_solution/server/lib/entity_analytics/risk_score/risk_weights.ts#L81-L100